kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,74 @@
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            '"%s" is similar to "%s". Try keeping them the same to reduce work for translators!',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    _dict_of_strings = {}
+    threshold = 0.8
+
+    @staticmethod
+    def clean_string(text):
+        """
+        This method removes the operators and other punctuations
+        used in the string to avoid unwanted data to add up to
+        the similarity between the strings.
+        """
+        for word in text:
+            if word not in string.punctuation:
+                text += "".join(word)
+        return text
+
+    @staticmethod
+    def marked_for_translation(funcname):
+        return funcname in ("_", "gettext_lazy")
+
+    def check_similar_string(self, translation_string):
+        cleaned_translation_string = self.clean_string(translation_string)
+        for key in self._dict_of_strings:
+            if (
+                levenshtein.normalized_similarity(
+                    cleaned_translation_string, self.clean_string(key)
+                )
+                >= self.threshold
+            ):
+                return key
+        return None
+
+    @utils.check_messages("similar-string")
+    def visit_call(self, node):
+        if not self.marked_for_translation(node.func.as_string()):
+            return
+
+        if not isinstance(node.args[0], astroid.nodes.Const):
+            return
+
+        translation_string = node.args[0].value
+        similar_string = self.check_similar_string(translation_string)
```

Comment:
`if translation_string in self._dict_of_strings` => return b/c we've seen this string before and it's exactly the same as it was so no need to calculate the similarity at all.